cmd/disk-cache-backend.go

```diff
@@ -45,6 +48,10 @@ const (
 	cacheMetaVersion  = "1.0.0"
 
 	cacheEnvDelimiter = ";"
+
+	// SSECacheEncrypted is the metadata key indicating that the object
+	// is a cache entry encrypted with cache KMS master key in globalCacheKMS.
+	SSECacheEncrypted = "X-Minio-Internal-Encrypted-Cache"
```

Comment:
Why is it necessary to mark a object (that has been cached) as "cache encrypted" - Isn't it sufficient to just get the cached object and try to decrypt it with the cache encryption key (KMS)?

Reply:
It is necessary to mark this object as cache encrypted so that decryptObjectInfo will get the object encryption key from globalCacheKMS instead of attempting to get it from globalKMS - since the OEK is determined at handler level, this marker is required.

Reply:
Yeah understood. Not super happy with that situation but I guess it's better to add the feature like this without massive code changes. Thanks.